mlibrary/pride
# ESLint fixes - Part 1

# Overview
In an effort to increase QA, all `./src` and `./spec` files (except the ones generated by `esbuild`) will have their linting errors fixed (according to `eslint`). Once all errors are fixed, `eslint` will be forced into every build.

This round of changes includes automated fixes, and manual fixes of all files that have existing unit tests written.

## Anything else?
The `no-undef` rule has been turned off for `./spec` files to remove the errors about `Pride` and `expect` not being defined. The future plan is to move testing off the browser and into Node.

## Testing
1. Build the repository (`npm run build`).
2. Open `./spec/index.html` in the browser to see if all tests pass.
   * Break the new/updated unit tests to make sure they are working properly.
3. Navigate to your local [Search repository](https://github.com/mlibrary/search) and apply the newly generated files:
   1. Open `./package.json` and add `#branch-name-of-pull-request` at the end of the `pride` dependency URL:
      ```bash
      "pride": "git+https://github.com/mlibrary/pride.git#branch-name-of-pull-request"
      ``` 
   2. Do a clean install of Search:
      ```bash
      rm -rf node_modules && rm package-lock.json && npm install
      ``` 
   3. Start Search (`npm start`) and look around [the site](http://localhost:3000/everything). Check to see if there are any console errors, and everything still works as expected.
